bittensor/_dataset/utils.py

```diff
@@ -0,0 +1,143 @@
+import threading
+import ctypes
+import time
+from typing import *
+import asyncio
+
+class CustomThread(threading.Thread):
```

Comment:
can you add a short docstring that explains exactly what functionality this adds?